src/Groups/directproducts.jl

```diff
@@ -278,6 +281,17 @@ function (G::SemidirectProductGroup{S,T})(a::GAPGroupElem{S},b::GAPGroupElem{T})
    return group_element(G,xgap)
 end
 
+"""
+    factor_of_semidirect_product(G::SemidirectProductGroup, j::Int)
+Return the `j`-th factor of `G`, for `j`=1,2.
+"""
+function factor_of_semidirect_product(G::SemidirectProductGroup, j::Base.Integer)
```

Comment:
I really don't like the names of these functions, I would not have guessed their meanings. Also, it seems weird to use an integer with just two allowed values as an index here. This would perhaps make some sense if there was a single function (with several methods) that can be applied to direct, semidirect and wreath products, but not so much when there is a dedicated function. Also, the documentation leaves it unclear which "factor" is which (i.e. is the normal subgroup returned for 1 or 2). If we really want dedicated functions, we might as well have two accessor functions for each of the two types (semidirect & wreath): one to get the "normal subgroup", one to get the "acting subgroups".

Reply:
The wreath product is the semidirect product of `H` and `NxNx...xN`. Does `normal_subgroup` return `N` or `NxNx...xN`? (the first makes much more sense, but the name of the function is technically wrong)